onnxruntime/core/graph/graph_utils.h

```diff
@@ -104,14 +108,14 @@ struct GraphEdge {
 
 #endif  // !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
 
-#if !defined(ORT_MINIMAL_BUILD)
 /** Checks if the node has the same operator since version as the given one. */
 bool MatchesOpSinceVersion(const Node& node, std::initializer_list<ONNX_NAMESPACE::OperatorSetVersion> versions);
```

Comment:
Do you need these functions as part of the minimal build? I assumed triton is not part of it.

Reply:
This is to fix the mismatch between h and cc file. In cc file, this function is outside "if !defined(ORT_MINIMAL_BUILD)" and is called by some code that is also outside "if !defined(ORT_MINIMAL_BUILD)".

Reply:
the change should be opposite. .i.e. move these under #if !defined(ORT_MINIMAL_BUILD) in .cc too.

The calling code is in optimizers which are not part of minimal build either.

Reply:
This is not consistent with IsSupportedOptypeVersionAndDomain, which is also avaible for minimal build, and call MatchesOpSinceVersion and MatchesOpSetDomain. Only node.Op() is not available on minimal build.

If we think IsSupportedOptypeVersionAndDomain is called by optimizers only (not sure it's true or not), then we need to move the whole IsSupportedOptypeVersionAndDomain under #if !defined(ORT_MINIMAL_BUILD). But I don't think we do this in the PR.

`
bool IsSupportedOptypeVersionAndDomain(const Node& node, std::string_view op_type,
                                       gsl::span<const ONNX_NAMESPACE::OperatorSetVersion> versions,
                                       std::string_view domain) {
  return (node.OpType() == op_type &&
  // we don't have op schemas in the minimal build so there's no way to check the deprecated flag
#if !defined(ORT_MINIMAL_BUILD)
          !node.Op()->Deprecated() &&
#endif
          MatchesOpSinceVersion(node, versions) && MatchesOpSetDomain(node, domain));
}
`